2016-2017_Starstruck/Module Test Code/FileIO_Test/include/AutonRecorder.h

```diff
@@ -0,0 +1,87 @@
+/*
+ *  AutonRecorder.h
+ *
+ *  Created on: Feb 8, 2017
+ *      Author: Kirsten Meidlinger
+ */
+
+#ifndef AUTONRECORDER_H_
+#define AUTONRECORDER_H_
+
+/*
+ * This is the library for 750E's Auton Recorder. It reads joystick values and writes them to
+ *   a file in the cortex's memory, and is able to read the file again to replay it as an
+ *   autonomous program. It can work for any game.
+ *
+ * This file should be placed in include and included in main.h
+ */
+
+
+  typedef struct Frame {
+      int analog_main[4]; //ch1, ch2, ch3, ch4
+      bool digital_main[12]; //5U, 5D, 6U, 6D, 7U, 7D, 7L, 7R, 8U, 8D, 8L, 8R
+      int analog_partner[4];
+      bool digital_partner[12];
+      struct Frame *next;
+      struct Frame *previous;
+    }Frame;
+
+
+  static const Frame NULLFRAME = {
+    {-1,-1,-1,-1},
```

Comment:
when defining structs, you want to try to use the more explicit syntax when possible. i.e.:

```
static const Frame NULLFRAME = {
    .analog_main = {-1, -1, -1, -1},
...
```

That way you can easily tell what's being assigned to what and you don't have to memorize the struct order